model/src/template/mod.rs

```diff
@@ -0,0 +1,553 @@
+                Token::Str("name"),
+                Token::Str("serialized_source_guild"),
+                Token::Struct {
+                    name: "TemplateGuild",
+                    len: 14,
+                },
+                Token::Str("afk_channel_id"),
+                Token::None,
+                Token::Str("afk_timeout"),
+                Token::U64(300),
+                Token::Str("channels"),
+                Token::Seq { len: Some(4) },
+                Token::Struct {
+                    name: "CategoryChannel",
+                    len: 5,
+                },
+                Token::Str("id"),
+                Token::NewtypeStruct { name: "ChannelId" },
+                Token::Str("1"),
+                Token::Str("type"),
+                Token::U8(4),
+                Token::Str("name"),
+                Token::Str("Text Channels"),
+                Token::Str("permission_overwrites"),
+                Token::Seq { len: Some(0) },
+                Token::SeqEnd,
+                Token::Str("position"),
+                Token::I64(0),
+                Token::StructEnd,
+                Token::Struct {
+                    name: "TextChannel",
+                    len: 8,
+                },
+                Token::Str("id"),
+                Token::NewtypeStruct { name: "ChannelId" },
+                Token::Str("2"),
+                Token::Str("type"),
+                Token::U8(0),
+                Token::Str("name"),
+                Token::Str("general"),
+                Token::Str("nsfw"),
+                Token::Bool(false),
+                Token::Str("parent_id"),
+                Token::Some,
+                Token::NewtypeStruct { name: "ChannelId" },
+                Token::Str("1"),
+                Token::Str("permission_overwrites"),
+                Token::Seq { len: Some(2) },
+                Token::Struct {
+                    name: "PermissionOverwriteData",
+                    len: 4,
+                },
+                Token::Str("allow"),
+                Token::Str("0"),
+                Token::Str("deny"),
+                Token::Str("2048"),
+                Token::Str("id"),
+                Token::Str("0"),
+                Token::Str("type"),
+                Token::U8(0),
+                Token::StructEnd,
+                Token::Struct {
+                    name: "PermissionOverwriteData",
+                    len: 4,
+                },
+                Token::Str("allow"),
+                Token::Str("2048"),
+                Token::Str("deny"),
+                Token::Str("0"),
+                Token::Str("id"),
+                Token::Str("2"),
+                Token::Str("type"),
+                Token::U8(0),
+                Token::StructEnd,
+                Token::SeqEnd,
+                Token::Str("position"),
+                Token::I64(0),
+                Token::Str("rate_limit_per_user"),
+                Token::Some,
+                Token::U64(0),
```

Comment:
If anyone wants to help, I can't figure out for the life of me why this test is failing. I think I've isolated this error to the text channel. Here's the error: 
```
thread 'template::tests::test_template' panicked at 'tokens failed to deserialize: invalid type: Option value, expected u64', /home/cassie/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_test-1.0.114/src/assert.rs:160:19
stack backtrace:
   0: rust_begin_unwind
             at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/std/src/panicking.rs:493:5
   1: std::panicking::begin_panic_fmt
             at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/std/src/panicking.rs:435:5
   2: serde_test::assert::assert_de_tokens
             at /home/cassie/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_test-1.0.114/src/assert.rs:160:19
   3: serde_test::assert::assert_tokens
             at /home/cassie/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_test-1.0.114/src/assert.rs:36:5
   4: twilight_model::template::tests::test_template
             at ./src/template/mod.rs:314:9
   5: twilight_model::template::tests::test_template::{{closure}}
             at ./src/template/mod.rs:50:5
   6: core::ops::function::FnOnce::call_once
             at /home/cassie/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:227:5
   7: core::ops::function::FnOnce::call_once
             at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/core/src/ops/function.rs:227:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```